src/cast_core/src/CollectionScanner.cpp

```diff
@@ -219,15 +234,22 @@ void collectionScan(CollectionScanner::PhaseConfig* config,
     bool scanFinished = false;
     auto statTracker = config->scanOperation.start();
     for (auto& collection : collections) {
-        auto filter = config->filterExpr ? config->filterExpr->evaluate()
-                                         : bsoncxx::document::view_or_value{};
-        auto docs = collection.find(session, filter, config->findOptions);
+        auto docs = [&]() {
+          if (config->aggregate) {
+              return collection.aggregate(session, evaluatePipeline(config), config->aggregateOptions);
+          } else {
+              auto filter = config->filterExpr ? config->filterExpr->evaluate()
+                                               : bsoncxx::document::view_or_value{};
+              return collection.find(session, filter, config->findOptions);
+          }
+        };
+
         /*
          * Try-catch this as the collection may have been deleted.
          * You can still do a find but it'll throw an exception when we iterate.
          */
         try {
```

Comment:
I think this code is dead code now. Shouldn't this be in the for loop?

Reply:
I don't get you here, this for loop is required to iteration over the cursor returned by the lambda.

Reply:
I got completely tripped up on the lambda and then somehow thought the previous context had been closed (it hadn't). My bad.